stream/src/main/java/org/apache/kafka/streaming/processor/internals/SinkNode.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streaming.processor.internals;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streaming.processor.ProcessorContext;
+
+public class SinkNode<K, V> extends ProcessorNode<K, V> {
+
+    private final String topic;
+    private final Serializer<K> keySerializer;
+    private final Serializer<V> valSerializer;
+
+    private ProcessorContext context;
+
+    public SinkNode(String name, String topic, Serializer<K> keySerializer, Serializer<V> valSerializer) {
+        super(name);
+
+        this.topic = topic;
+        this.keySerializer = keySerializer;
+        this.valSerializer = valSerializer;
+    }
+
+    @Override
+    public void addChild(ProcessorNode<?, ?> child) {
+        throw new UnsupportedOperationException("sink node does not allow addChild");
+    }
+
+    @Override
+    public void init(ProcessorContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public void process(K key, V value) {
+        // send to all the registered topics
+        RecordCollector collector = ((ProcessorContextImpl) context).recordCollector();
+        collector.send(new ProducerRecord<>(topic, key, value), keySerializer, valSerializer);
```

Comment:
The casting done on line 53 from `ProcessorContext` to `ProcessorContextImpl` makes testing difficult, as the tests would require a `ProcessingContextImpl` rather than an alternative. (See how `KStreamTestDriver` uses a `MockProcessorContext` for an example.) I assume that `recordCollector()` was removed from `ProcessorContext` to hide it from `Processor` implementations, so this cast can either be kept and the test classes required to use a subclass of `ProcessorContextImpl`, or the `ProcessorContextImpl` implements another internal interface that `SinkNode` can use here when casting and the test classes can choose to implement. Thoughts?

(It looks like tests that use `KStreamTestDriver` do so by implementing a tail-end mock processor. I'm trying to create a similar test driver for testing a `Processor` and a `Topology`. Right now the only problem is that the `SinkNode` throws a ClassCastException on line 53.)